src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRunCodegenNodeFactory.cs

```diff
@@ -25,6 +24,29 @@ public interface IEETypeNode
         TypeDesc Type { get; }
     }
 
+    public struct NodeCache<TKey, TValue>
+    {
```

Comment:
For the future, it would be good to not do this kind of refactoring (moving blocks of code around while making some changes to them) when the PR contains other numerous changes. It makes it easier to review.
For instance, I see that you have removed the `GetOrAdd(TKey key, Func<TKey, TValue> creator)` function, but that wasn't easy to spot